src/mainwindow.cpp

```diff
@@ -109,6 +114,34 @@ void MainWindow::showTabAndTop() {
     getTopWidget()->show();
 }
 
+void MainWindow::updateTabButtons()
+{
+    auto tabBar = getTabBar();
+    QRect tabBarRect = getTabBar()->rect();
+    QRect newButtonTabRect = tabBar->tabRect(tabBar->count() - 1);
+
+    /* Decision is made at half way of the new button tab for smoothness */
+    newButtonTabRect.setWidth(newButtonTabRect.width() / 2);
+    bool newTabVisible = tabBarRect.contains(newButtonTabRect);
+    if (mp_ui->newTabSideButton->isHidden())
+        mp_ui->newTabSideButton->setVisible(!newTabVisible);
+    else
+        mp_ui->newTabSideButton->setHidden(newTabVisible);
```

Comment:
Can't the `if/else` statement be reduced to `mp_ui->newTabSideButton->setVisible(!newTabVisible);`?

Reply:
Done